library/src/scala/compiletime/ops/package.scala

```diff
@@ -0,0 +1,174 @@
+package scala.compiletime
+
+import scala.annotation.infix
+
+package object ops {
+  object any {
+    /** Equality comparison of two singleton types.
+     * ```scala
+     * val eq1: 1 == 1 = true
+     * val eq2: 1 == "1" = false
+     * val eq3: "1" == "1" = true
+     * ```
+     */
+    @infix type ==[X <: AnyVal, Y <: AnyVal] <: Boolean
+
+    /** Inequality comparison of two singleton types.
+     * ```scala
+     * val eq1: 1 != 1 = false
+     * val eq2: 1 != "1" = true
+     * val eq3: "1" != "1" = false
+     * ```
+     */
+    @infix type !=[X <: AnyVal, Y <: AnyVal] <: Boolean
+  }
+
+  object string {
+    /** Concatenation of two `String` singleton types.
+     *  ```scala
+     *  val hello: "hello " + "world" = "hello world"
+     *  ```
+     */
+    @infix type +[X <: String, Y <: String] <: String
+  }
+
+  object int {
+    /** Addition of two `Int` singleton types. Type-level equivalent of `scala.Int.+`
```

Comment:
I would remove all the `Type-level equivalant` comments since it's pretty obvious. Also it's inaccurate as `+` is not actually a method on `object Int`...

Reply:
All right, fixed in 7e0a1db10f4a1e35872fa04c2426ede2b6a6a004.